serial_server.py

```diff
@@ -31,8 +31,11 @@
 ### END NODE INFO
 """
 
+import sys
+import os
 from time import sleep
 import time
+import collections
```

Comment:
imports should be sorted. I would also suggest removing the bare import of `sleep` and just referring to `time.sleep`, though that may be out of scope for this change.